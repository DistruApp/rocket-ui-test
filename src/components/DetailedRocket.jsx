import React from 'react';
import PropTypes from 'prop-types';

const DetailedRocket = ({ rocket }) => (
  <div>
    <div> Rocket ID: { rocket.id } </div>
    <div> Rocket Cost per launch: { rocket.cost_per_launch } </div>
    <div> Rocket Description: { rocket.description } </div>
  </div>
);
DetailedRocket.propTypes = {
  rocket: PropTypes.shape(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      cost_per_launch: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired
};

export default DetailedRocket
