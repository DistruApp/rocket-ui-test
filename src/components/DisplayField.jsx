import React from 'react';
import PropTypes from 'prop-types';

const DisplayField = ({ label, value }) => <p style={{ padding: '4px 0' }}><strong>{label}:</strong>&nbsp;{value}</p>

DisplayField.propTypes = {
  label: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
}

export default DisplayField;