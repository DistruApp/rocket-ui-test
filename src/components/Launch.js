import React from 'react';

const Launch = (props) => {
	const { launch } = props;

	return (
		<li>
			<h2> {launch.mission_name} </h2>
			<div> Flight Number: {launch.flight_number} </div>
		</li>
	);
};

export default Launch;
