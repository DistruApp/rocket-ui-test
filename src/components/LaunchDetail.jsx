import React, {useContext} from 'react';
import { store } from "../stores/store";

const Launch = ({ launch }) => {
    const { dispatch, state } = useContext(store)

    const rocket = state.rockets.filter(rocket => rocket.id === launch.rocket)[0]

    return (
        <>
            <div className="rocket-detail">
                    {
                        launch.rocket && rocket &&
                            <>
                                <div>
                                    <h5>
                                        Rocket Name
                                    </h5>
                                    <span>
                                        { rocket.name }
                                    </span>
                                    <h5>
                                        Rocket ID
                                    </h5>
                                    <span>
                                        { launch.rocket }
                                    </span>
                                </div>
                                <div>
                                    <h5>
                                        Cost per Launch
                                    </h5>
                                    <span>
                                        { `$${rocket.cost_per_launch.toLocaleString()}` }
                                    </span>
                                </div>
                            </>
                    }
            </div>
            <div className="launch-detail">
                {
                    launch.details &&
                        <div>
                            <h5>
                                Details
                            </h5>
                            <span>
                                { launch.details }
                            </span>
                        </div>
                }
            </div>
        </>
    );
}

export default Launch;
