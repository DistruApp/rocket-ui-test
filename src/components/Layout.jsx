import React from 'react';
import PropTypes from 'prop-types';

export const Layout = ({children}) => (
  <main>
    <h3>
      Sinan Talha Rocket-Ui Test
    </h3>
    <section>
      { children }
    </section>
  </main>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};