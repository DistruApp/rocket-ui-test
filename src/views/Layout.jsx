import React from 'react';
import Navigation from "../components/Navigation";

const Layout = ({ children }) => {

  return (
    <main className={`layout`}>
        <Navigation/>
        <section>
            { children }
        </section>
    </main>
  );
};

export default Layout;
