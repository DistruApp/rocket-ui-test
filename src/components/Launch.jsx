import React from 'react';
import { connect } from "react-redux";

import { shouldFetchRocket, fetchRocket } from "../actions/Rockets";
import { getRocket } from "../queries/Rockets";
import DetailedRocket from "./DetailedRocket";

const Launch = ({ launch, active, rocket, dispatch }) => {
  const rocketId = launch.rocket;
  // Fetching in this component so that when fetch occurs, only this component
  // updates and not all the other launches
  if (active && !rocket) fetchRocket(dispatch, rocketId);

  return (
    <div>
      <h2> { launch.name } </h2>
      <div> Flight Number: { launch.flight_number } </div>
      { active && rocket && <DetailedRocket rocket={rocket} /> }
    </div>
  );
}

const mapStateToProps = ({ rocketCollection }, { launch: { rocket: rocketId } }) => {
  const rocket = getRocket(rocketCollection, rocketId);
  return { rocket }
};
const mapDispatchToProps = dispatch => ({ dispatch });

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Launch);
