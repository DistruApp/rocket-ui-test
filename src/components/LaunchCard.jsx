/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import clsx from 'clsx';
import DisplayField from './DisplayField'

const LaunchCard = ({ launch = {}, launchOnClick, selected }) => {
  const { mission_name, flight_number, launch_success } = launch;
  const missionPatch = _.get(launch, 'links.mission_patch_small');


  return <li className={clsx('launch-card', { selected })} onClick={() => launchOnClick(launch)} role="button">
    <div className="launch-card-image">
      {missionPatch ? <img src={missionPatch} alt={`${mission_name}`} /> : <div className="mission-patch-not-found">MISSION PATCH NOT AVAILABLE</div>}
    </div>
    <div className="launch-card-detail">
      <div>
        <h3> {mission_name} </h3>
      </div>
      <DisplayField label="Flight Number" value={flight_number} />
      <div>
        <h5 className={clsx({ 'text-success': launch_success, 'text-error': !launch_success })} >MISSION {launch_success ? 'SUCCESS' : 'FAILURE'}</h5>
      </div>
    </div>
  </li>
}

LaunchCard.propTypes = {
  launch: PropTypes.object.isRequired,
  launchOnClick: PropTypes.func.isRequired,
  selected: PropTypes.bool.isRequired,
}

export default LaunchCard;
