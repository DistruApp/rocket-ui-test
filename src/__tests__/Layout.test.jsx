import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Layout } from '../components/Layout';

configure({ adapter: new Adapter() });

describe('Layout component test', () => {
  it('renders Layout component without crashing', () => {
    const component = mount(
      <Layout>
        <div>test</div>
      </Layout>
    );
    expect(component.getElements()).toMatchSnapshot();
  });
});
