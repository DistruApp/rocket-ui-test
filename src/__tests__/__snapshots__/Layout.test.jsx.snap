// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component test renders Layout component without crashing 1`] = `
Array [
  <Layout>
    <div>
      test
    </div>
  </Layout>,
]
`;
